code/formfields/UserFormsNumericField.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+
+class UserFormsNumericField extends NumericField
+{
+
+    private $min;
+
+    private $max;
+
+    /**
+     * @return mixed
+     */
+    public function getMin()
+    {
+        return $this->min;
+    }
+
+    /**
+     * @param mixed $min
+     */
+    public function setMin($min)
+    {
+        if (!is_numeric($min)) {
+            throw new Exception('$min must be a number.');
+        }
+
+        $this->min = $min;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getMax()
+    {
+        return $this->max;
+    }
+
+    /**
+     * @param mixed $max
+     */
+    public function setMax($max)
+    {
+        if (!is_numeric($max)) {
+            throw new Exception('$min must be a number.');
+        }
+        $this->max = $max;
+    }
+
+    public function validate($validator)
+    {
+        $isValid = parent::validate($validator);
+        if (!$isValid) {
+            return false;
+        }
+
+        if ((!empty($this->min) && $this->value < $this->min) || (!empty($this->max) && $this->value > $this->max)) {
+            $validator->validationError(
+                $this->name,
+                sprintf("This must be between %s and %s", $this->min, $this->max),
```

Comment:
This needs to be translatable please.
You can [read about this](https://docs.silverstripe.org/en/4/developer_guides/i18n/#translating-text) if you are unsure how :)

Reply:
@NightJar Right, i've updated the pull request.